src/main/java/io/vertx/core/http/impl/WebSocketImplBase.java

```diff
@@ -56,25 +57,37 @@
   protected Handler<Void> closeHandler;
   protected Handler<Void> endHandler;
   protected boolean closed;
-
+  private EventBus eventBus;
   protected WebSocketImplBase(VertxInternal vertx, ConnectionBase conn, boolean supportsContinuation,
                               int maxWebSocketFrameSize) {
     this.supportsContinuation = supportsContinuation;
-    this.textHandlerID = UUID.randomUUID().toString();
-    this.binaryHandlerID = UUID.randomUUID().toString();
     this.conn = conn;
-    Handler<Message<Buffer>> binaryHandler = msg -> writeBinaryFrameInternal(msg.body());
-    binaryHandlerRegistration = vertx.eventBus().<Buffer>localConsumer(binaryHandlerID).handler(binaryHandler);
-    Handler<Message<String>> textHandler = msg -> writeTextFrameInternal(msg.body());
-    textHandlerRegistration = vertx.eventBus().<String>localConsumer(textHandlerID).handler(textHandler);
+    this.eventBus = vertx.eventBus();
     this.maxWebSocketFrameSize = maxWebSocketFrameSize;
   }
-
   public String binaryHandlerID() {
+	  return binaryHandlerID(false);
+  }
+  
+  public synchronized String binaryHandlerID(boolean cluster) {
+    if(binaryHandlerID == null){
```

Comment:
cosmetics, please use appropriate whitespace (see code style guidelines on the wiki)